main/api.c

```diff
@@ -329,6 +330,43 @@ static void read_cb(evutil_socket_t sock, short what, void * /*priv*/) {
 		event_free_finalize(0, ev, finalize_fd);
 }
 
+static int check_permission(int fd) {
+	cap_flag_value_t has_cap = CAP_CLEAR;
+	struct ucred cred;
+	socklen_t len = sizeof(cred);
+	cap_t caps;
+
+	if (getsockopt(fd, SOL_SOCKET, SO_PEERCRED, &cred, &len) == -1) {
+		LOG(ERR, "getsockopt(SO_PEERCRED): %s", strerror(errno));
+		return -1;
+	}
+
+	// root user, allowed
+	if (cred.uid == 0)
+		return 0;
+
+	// check if peer process has CAP_NET_ADMIN
+	caps = cap_get_pid(cred.pid);
+	if (caps == NULL) {
+		LOG(ERR, "cap_get_pid: %s", strerror(errno));
+		return -1;
+	}
+
+	if (cap_get_flag(caps, CAP_NET_ADMIN, CAP_EFFECTIVE, &has_cap) == -1) {
+		LOG(ERR, "cap_get_flags: %s", strerror(errno));
+		cap_free(caps);
+		return -1;
+	}
+	cap_free(caps);
+
+	if (has_cap != CAP_SET) {
```

Comment:
I would prefer if the check was more dynamic.

* If the `grout` process is started as `root` or as any user that has the `CAP_NET_ADMIN` capability, then require that all clients have that same capability.
* Else, if `grout` is started as a regular user without `CAP_NET_ADMIN`, then require that client processes are started with the same `gid` than the `grout` process.